impl/src/bin/cargo-raze.rs

```diff
@@ -96,83 +105,85 @@ fn main() -> Result<()> {
     lock_path_opt,
   };
 
+  // Determine the path to the Bazel workspace root or fallback to the current working directory
```

Comment:
Purely aesthetic thing, but it looks like your inline comments here are meant to describe blocks of code, but there are newlines sprinkled throughout which makes the association not obvious.

If 
```rust
 // Determine the path to the Bazel workspace root or fallback to the current working directory
```
is meant to describe L109 - L117, I think you should remove the newlines:

```rust
  let cargo_raze_working_dir = find_bazel_workspace_root().unwrap_or(std::env::current_dir()?);
  let remote_genmode_inputs = gather_remote_genmode_inputs(&cargo_raze_working_dir, &settings);
  let metadata = metadata_fetcher.fetch_metadata(
    &files,
    remote_genmode_inputs.binary_deps,
    remote_genmode_inputs.override_lockfile,
  )?;
```

The same applies to the `// Do Planning` comment, as well as the `// Render BUILD files` comment, as well as the `// Write BUILD files` comment.

Reply:
Do you think that comment is still necessary with the renamed comment? I really felt the `// Section ---` style clearly communicated that the `main` method was sectional and made those comments distinct from others describing functionality

Reply:
The essential aspect of my comment is that "// some comment" doesn't obviously apply to multiple lines of code once you insert newlines between the lines of code.

The "// whatever ---" seems completely arbitrary. Arbitrary conventions don't really communicate intent very well.

Reply:
Ah, I see. So what's the action item from here on?

Reply:
I'm saying don't do the "---" thing. When a comment applies to multiple lines, you have two options.

1. Keep the lines of code together. Don't add whitespace
2. Move the lines into a function.